revolt/ext/commands/cog.py

```diff
@@ -33,7 +33,17 @@ class Cog(metaclass=CogMeta):
     _commands: list[Command]
     qualified_name: str
 
+    def cog_load(self):
+        """A special method that is called when the cog gets loaded."""
+        pass
+
+    def cog_unload(self):
+        """A special method that is called when the cog gets removed."""
+        pass
+
     def _inject(self, client: CommandsClient):
+        self.cog_load()
```

Comment:
this should be moved to the end of the function

Reply:
I think putting at the current place is better because we won't have to roll back the whole inject process if something goes wrong inside `cog_load`.